### PR TITLE
[docs] Add checking of all links in documentation and fix broken links

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

As per PR #1786 there continues to be broken links, this updates the `docs_test` tool to check all links, not just those with `https://github.com/...` prefixes, and also local repository paths.

Example (this already runs in CI, just updated to check all link types):
```bash
$ make docs-validate
mkdir -p /Users/r/go/src/github.com/m3db/m3/bin
Building docs_test
go build -o /Users/r/go/src/github.com/m3db/m3/bin/docs_test ./src/cmd/tools/docs_test/main/.
./bin/docs_test
2019/07/02 15:23:42 validated docs (repoPathsValidated=394)
```

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: docs/how_to/kubernetes.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 **Please note:** If possible _[PLEASE USE THE OPERATOR](https://operator.m3db.io/)_ to deploy to Kubernetes if you
 can. It is a considerly more streamlined setup.
 
-The [operator](https://operator.m3db.io/) leverages [custom resource definitions](https://v1-10.docs.kubernetes.io/docs/concepts/api-extension/custom-resources/)
+The [operator](https://operator.m3db.io/) leverages [custom resource definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
```
Comment (benraskin92) [inline docs/how_to/kubernetes.md]: Should we keep these and just update the version?

File: src/cmd/tools/docs_test/main/main.go [hunk 2]
```diff
@@ -75,15 +78,48 @@ func main() {
...
+				if httpOrHTTPS(resolvedPath) {
+					// Check external link using HEAD request
+					client := &http.Client{
+						Transport: &http.Transport{
+							// Some sites have invalid certs, like some kubernetes docs
+							TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+						},
+					}
+					resp, err := client.Head(resolvedPath)
+					if err == nil && resp.StatusCode/100 != 2 {
```
Comment (benraskin92) [inline src/cmd/tools/docs_test/main/main.go]: Haha nice. Would be neat if `http.Response` had like an `isSuccessful` method or something.

File: src/cmd/tools/docs_test/main/main.go [hunk 3]
```diff
@@ -100,3 +136,20 @@ func main() {
...
 	log.Printf("validated docs (repoPathsValidated=%d)\n", repoPathsValidated)
 }
+
+func httpOrHTTPS(url string) bool {
+	return strings.HasPrefix(strings.ToLower(url), "http://") ||
+		strings.HasPrefix(strings.ToLower(url), "https://")
+}
+
+type checkedLink struct {
+	Document     string
```
Comment (benraskin92) [inline src/cmd/tools/docs_test/main/main.go]: Do these need to be exported?

File: src/cmd/tools/docs_test/main/main.go [hunk 4]
```diff
@@ -100,3 +136,20 @@ func main() {
 
 	log.Printf("validated docs (repoPathsValidated=%d)\n", repoPathsValidated)
 }
+
+func httpOrHTTPS(url string) bool {
```
Comment (benraskin92) [inline src/cmd/tools/docs_test/main/main.go]: nit: rename to `isHTTPOrHTTPS`

File: docs/how_to/kubernetes.md (hunk 1 above)
Comment (robskillington) [reply docs/how_to/kubernetes.md]: They've completely moved it unfortunately, I can't get access to any versioned docs.

File: src/cmd/tools/docs_test/main/main.go (hunk 2 above)
Comment (robskillington) [reply src/cmd/tools/docs_test/main/main.go]: Yeah lol... sadly not.

File: src/cmd/tools/docs_test/main/main.go (hunk 3 above)
Comment (robskillington) [reply src/cmd/tools/docs_test/main/main.go]: probably not, will unexport.

**Pull request merged**